Shopify/cli
# Rename presets to environments

### WHY are these changes introduced?

Fixes https://github.com/Shopify/internal-cli-foundations/issues/541

### WHAT is this pull request doing?

- Renames everything related to presets to environments, including the current file name (`shopify.environments.toml`) and flag (`--environment`).
- Rename the previous `environment` concept (like `ensureDevEnvironment`) to `context` to avoid confusion

### How to test your changes?

Create a file called `shopify.environments.toml` in the root folder of an app with the content:
```
[test]
json = true
```
And then run `yarn shopify app info --environment test`. It should show that the environment is being used, and output the info in json format.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
